cpp/test/packed/stream_writer_test.cpp

```diff
@@ -0,0 +1,139 @@
+// Copyright 2024 Zilliz
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <arrow/filesystem/localfs.h>
+#include <gtest/gtest.h>
+#include <arrow/api.h>
+#include <parquet/api/writer.h>
+#include <packed/stream_writer.h>
+#include <parquet/properties.h>
+#include <common/status.h>
+#include <packed/reader.h>
+#include <filesystem>
+#include <memory>
+#include <filesystem>
+#include <arrow/type.h>
+#include <arrow/type_fwd.h>
+#include <arrow/array/builder_binary.h>
+#include <arrow/array/builder_primitive.h>
+#include <arrow/util/key_value_metadata.h>
+#include "gmock/gmock.h"
+#include "packed/reader.h"
+#include "test_util.h"
+#include "arrow/table.h"
+
+using namespace std;
+
+namespace milvus_storage {
+
+class StreamWriterTest : public ::testing::Test {
+  protected:
+  StreamWriterTest() : props_(*parquet::default_writer_properties()) {}
+
+  void SetUp() override {
+    file_path_ = std::filesystem::temp_directory_path() / "stream_writer_test";
+    std::filesystem::create_directory(file_path_);
```

Comment:
There is a `BuildFileSystem` call in fs_util.h, which returns a `arrow::fs::FileSystem`. We'd better use that call. I suppose we won't use `std::filesystem` anywhere in this project.